alerting/src/main/kotlin/org/opensearch/alerting/util/SupportedApiSettingsExtensions.kt

```diff
@@ -0,0 +1,134 @@
+        )
+        is ClusterStatsResponse -> redactFieldsFromResponse(
+            this.convertToMap(),
+            SupportedApiSettings.getSupportedJsonPayload(ApiType.CLUSTER_STATS.defaultPath)
+        )
+        is ClusterGetSettingsResponse -> redactFieldsFromResponse(
+            this.convertToMap(),
+            SupportedApiSettings.getSupportedJsonPayload(ApiType.CLUSTER_SETTINGS.defaultPath)
+        )
+        is NodesStatsResponse -> redactFieldsFromResponse(
+            this.convertToMap(),
+            SupportedApiSettings.getSupportedJsonPayload(ApiType.NODES_STATS.defaultPath)
+        )
+        is PendingClusterTasksResponse -> redactFieldsFromResponse(
+            this.convertToMap(),
+            SupportedApiSettings.getSupportedJsonPayload(ApiType.CAT_PENDING_TASKS.defaultPath)
+        )
+        is RecoveryResponse -> redactFieldsFromResponse(
+            this.convertToMap(),
+            SupportedApiSettings.getSupportedJsonPayload(ApiType.CAT_RECOVERY.defaultPath)
+        )
+        is GetRepositoriesResponse -> redactFieldsFromResponse(
+            this.convertToMap(),
+            SupportedApiSettings.getSupportedJsonPayload(ApiType.CAT_REPOSITORIES.defaultPath)
+        )
+        is GetSnapshotsResponse -> redactFieldsFromResponse(
+            this.convertToMap(),
+            SupportedApiSettings.getSupportedJsonPayload(ApiType.CAT_SNAPSHOTS.defaultPath)
+        )
+        is ListTasksResponse -> redactFieldsFromResponse(
+            this.convertToMap(),
+            SupportedApiSettings.getSupportedJsonPayload(ApiType.CAT_TASKS.defaultPath)
+        )
+        else -> throw IllegalArgumentException("Unsupported ActionResponse type: ${this.javaClass.name}")
+    }
+}
+
+/**
+ * Populates a [HashMap] with only the values that support being exposed to users.
+ * @param mappedActionResponse The response from the [LocalUriInput] API call.
+ * @param supportedJsonPayload The JSON payload as configured in [SupportedApiSettings.RESOURCE_FILE].
+ * @return The response values [HashMap] without the redacted fields.
+ */
+@Suppress("UNCHECKED_CAST")
+fun redactFieldsFromResponse(
+    mappedActionResponse: Map<String, Any>,
+    supportedJsonPayload: Map<String, ArrayList<String>>
+): Map<String, Any> {
+    return when {
+        supportedJsonPayload.isEmpty() -> mappedActionResponse
+        else -> {
+            val output = hashMapOf<String, Any>()
+            for ((key, value) in supportedJsonPayload) {
+                when (val mappedValue = mappedActionResponse[key]) {
+                    is Map<*, *> -> output[key] = XContentMapValues.filter(
+                        mappedActionResponse[key] as MutableMap<String, *>?,
+                        value.toTypedArray(), arrayOf()
+                    )
+                    else -> output[key] = mappedValue ?: hashMapOf<String, Any>()
+                }
+            }
+            output
+        }
+    }
```

Comment:
A few possible concerns with this format I see:

1.  We are using `supported_json_payloads.json` to retrieve the allowed keys. This implies that whatever is explicitly mentioned is what is supported, however, if the map in the json is empty (`{}`) we allow everything through even though that would imply that nothing is supported.
2. Do we support redaction of a nested map of allowed keys for more complex responses (assuming the result is not flattened)?

Reply:
These are good points.
1. I can see how this could be a bit confusing. For this use case, not including the API in the JSON would be the best way to redact everything from its response. This will be something we want to explain well in the documentation.
2. If I'm understanding this point correctly, yes. The `test redactFieldsFromResponse with non-empty supportedJsonPayload` test case examines that functionality.